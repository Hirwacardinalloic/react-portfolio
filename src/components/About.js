import Loic from '../img/Loic.png';
const About = () => {
    return ( 
        <div className="about">
            
            <section className='section1'>
                <p><b>Hello, my name is Loic Cardinal Hirwa</b>.
                I am a software developper, I started to know this
                field when I was young but I started digging in it 
                in 2019 when I found out that this field is very 
                intersting, helpful and enjoyable. That's why I 
                took a decision of empowering future software 
                developpers and feel confident every single day
                in this field.</p>
                <img src={Loic} alt='Loic'/>
            </section>
            <section className='section2'> 
                <p><b>I love what I do.</b> 
                As a software developper am 
                here to make you feel good</p>
            </section>
            <section className="section3">
            <h1>MY VALUES:</h1>
                <div>
                    <button>Transparency</button>
                   <p>To be true  to my clients by providing 
                    solutions to them</p>
                   <button>Commitment</button>
                   <p>Loyalty and hard work over everything
                    else.</p>
                   <button>Self improvement</button>
                   <p>Never stop learning,never stop
                    evolving into who i want to be and
                    where i want to be</p>
                
                </div>
            </section>
            <section className="section4">
                <h1>MY MISSION:</h1>
                    <p>My mission is to make software developer feel empowering 
                        and self confindent every single day </p>
            </section>
        </div>
     );
}
 
export default About;