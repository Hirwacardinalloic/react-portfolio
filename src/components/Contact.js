const Contact = () => {
    return ( 
        <div className="contact">
            <p>Fill in the form and submit it so that we get your message</p>
            <form>
                <label>Name</label>
                <input type='text' placeholder='Names' required/>
                <label>Email</label>
                <input type='email' placeholder='example@example.com' required/>
                <label>Message</label>
                <textarea placeholder='Write your message here' required></textarea>
                <button>Send message</button>
            </form>
        </div>
     );
}
 
export default Contact;