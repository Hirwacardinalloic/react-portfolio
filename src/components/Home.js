import Loic from '../img/Loic.png';
import html from '../img/html.png';
import javascript from '../img/javascript.png';
import css from '../img/css.png';
const Home = () => {
    return ( 
        <div className="home">
            <div className='introduction'>
                <p>Welcome to my portfolio, I am Hirwa Cardinal Loic a software developer.</p>
                <img src={Loic} alt='Loic'/>
            </div>
            <div className="skills">
                <h1>Skills</h1>
                <div className='images'>
                    <img src={html} alt='html'/>
                    <img src={css} alt='css'/>
                    <img src={javascript} alt='javascript'/>
                </div>

            </div>
        </div>
     );
}
 
export default Home;