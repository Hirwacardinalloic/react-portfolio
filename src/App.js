import NavBar from './components/layout/Navbar';
import { Route,BrowserRouter as Router, Switch } from 'react-router-dom';
import Home from './components/Home';
import About from './components/About';
import Contact from './components/Contact';
import Footer from './components/layout/Footer';
import CreateBlog from './components/blogs/CreateBlog';
import NotFound from './components/NotFound';
import BlogList from './components/blogs/BlogList';
import BlogDetails from './components/blogs/BlogDetails';
import Login from './components/auth/Login';
import SignUp from './components/auth/SignUp';



function App() {
  return (
    <Router>
       <div className="App">
          <NavBar />
          <div className="content">
            <Switch>
              <Route exact path='/'>
                <Home />
              </Route>
              <Route exact path='/about'>
                <About />
              </Route>
              <Route exact path='/contact'>
                <Contact />
              </Route>
              <Route exact path='/create'>
                <CreateBlog />
              </Route>
              <Route exact path='/blogs'>
                <BlogList />
              </Route>
              <Route exact path='/blogs/:id' component={BlogDetails} />
              <Route exact path='/login' component={Login} />
              <Route exact path='/signup' component={SignUp} />
              <Route path='*'>
                <NotFound />
              </Route>
            </Switch>

          </div>
          <Footer />
       </div>
    </Router>
   
  );
}

export default App;
