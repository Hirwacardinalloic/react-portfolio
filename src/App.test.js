import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import About from './components/About';
import Home from './components/Home';
import Contact from './components/Contact';
import NotFound from './components/NotFound';
import toJson from 'enzyme-to-json';
describe("Rendering components", ()=>{
    it('renders App component without crashing', ()=>{
        shallow(<App />)
    });
    it('renders About component without crashing', ()=>{
        shallow(<About />)
    });
    it('renders Home component without crashing', ()=>{
        shallow(<Home />)
    });
    it('renders Contact component without crashing', ()=>{
        shallow(<Contact />)
    });
    it('renders NotFound component without crashing', ()=>{
        shallow(<NotFound />)
    });
});

describe("SnapShots", ()=>{
    it('App snapshots', ()=>{
        const appTree = shallow(<App />);
        expect(toJson(appTree)).toMatchSnapshot();
    });
    it('About snapshots', ()=>{
        const aboutTree = shallow(<About />);
        expect(toJson(aboutTree)).toMatchSnapshot();
    });
    it('Home snapshots', ()=>{
        const homeTree = shallow(<Home />);
        expect(toJson(homeTree)).toMatchSnapshot();
    });
    it('Contact snapshots', ()=>{
        const contactTree = shallow(<Contact />);
        expect(toJson(contactTree)).toMatchSnapshot();
    });
    it('NotFound snapshots', ()=>{
        const notFoundTree = shallow(<NotFound />);
        expect(toJson(notFoundTree)).toMatchSnapshot();
    });
})