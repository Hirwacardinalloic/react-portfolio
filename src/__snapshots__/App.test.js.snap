// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShots About snapshots 1`] = `
<div
  className="about"
>
  <section
    className="section1"
  >
    <p>
      <b>
        Hello, my name is Loic Cardinal Hirwa
      </b>
      . I am a software developper, I started to know this field when I was young but I started digging in it in 2019 when I found out that this field is very intersting, helpful and enjoyable. That's why I took a decision of empowering future software developpers and feel confident every single day in this field.
    </p>
    <img
      alt="Loic"
      src="Loic.png"
    />
  </section>
  <section
    className="section2"
  >
    <p>
      <b>
        I love what I do.
      </b>
      As a software developper am here to make you feel good
    </p>
  </section>
  <section
    className="section3"
  >
    <h1>
      MY VALUES:
    </h1>
    <div>
      <button>
        Transparency
      </button>
      <p>
        To be true  to my clients by providing solutions to them
      </p>
      <button>
        Commitment
      </button>
      <p>
        Loyalty and hard work over everything else.
      </p>
      <button>
        Self improvement
      </button>
      <p>
        Never stop learning,never stop evolving into who i want to be and where i want to be
      </p>
    </div>
  </section>
  <section
    className="section4"
  >
    <h1>
      MY MISSION:
    </h1>
    <p>
      My mission is to make software developer feel empowering and self confindent every single day 
    </p>
  </section>
</div>
`;

exports[`SnapShots App snapshots 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <Connect(Navbar) />
    <div
      className="content"
    >
      <Switch>
        <Route
          exact={true}
          path="/"
        >
          <Home />
        </Route>
        <Route
          exact={true}
          path="/about"
        >
          <About />
        </Route>
        <Route
          exact={true}
          path="/contact"
        >
          <Contact />
        </Route>
        <Route
          exact={true}
          path="/create"
        >
          <Connect(CreateBlog) />
        </Route>
        <Route
          exact={true}
          path="/blogs"
        >
          <BlogList />
        </Route>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "type": [Function],
            }
          }
          exact={true}
          path="/blogs/:id"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "type": [Function],
            }
          }
          exact={true}
          path="/login"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "type": [Function],
            }
          }
          exact={true}
          path="/signup"
        />
        <Route
          path="*"
        >
          <NotFound />
        </Route>
      </Switch>
    </div>
    <Footer />
  </div>
</BrowserRouter>
`;

exports[`SnapShots Contact snapshots 1`] = `
<div
  className="contact"
>
  <p>
    Fill in the form and submit it so that we get your message
  </p>
  <form>
    <label>
      Name
    </label>
    <input
      placeholder="Names"
      required={true}
      type="text"
    />
    <label>
      Email
    </label>
    <input
      placeholder="example@example.com"
      required={true}
      type="email"
    />
    <label>
      Message
    </label>
    <textarea
      placeholder="Write your message here"
      required={true}
    />
    <button>
      Send message
    </button>
  </form>
</div>
`;

exports[`SnapShots Home snapshots 1`] = `
<div
  className="home"
>
  <div
    className="introduction"
  >
    <p>
      Welcome to my portfolio, I am Hirwa Cardinal Loic a software developer.
    </p>
    <img
      alt="Loic"
      src="Loic.png"
    />
  </div>
  <div
    className="skills"
  >
    <h1>
      Skills
    </h1>
    <div
      className="images"
    >
      <img
        alt="html"
        src="html.png"
      />
      <img
        alt="css"
        src="css.png"
      />
      <img
        alt="javascript"
        src="javascript.png"
      />
    </div>
  </div>
</div>
`;

exports[`SnapShots NotFound snapshots 1`] = `
<div>
  <h2>
    404 oops page not found!!!
  </h2>
</div>
`;
